### PR TITLE
Fix reading PLAIN_DICTIONARY field

### LINKED ISSUE #10: Issue when reading PLAIN_DICTIONARY fields
I am currently working on Usage/Examples section in README and wanted to add couple of code snippets on how to use library (I hope you do not object to this). While working on it, found that there is a potential issue when reading columns with PLAIN_DICTIONARY encoding.

I am trying to read `data/alltypes_plain.snappy.parquet` file that is in repository with code that is attached below.

Discovered that this returns me empty buffers when I read values. For example, the first column in Parquet file is "id" that has values `6` and `7`:
```
column 0:
--------------------------------------------------------------------------------
column type: INT32
column path: "id"
encodings: RLE PLAIN_DICTIONARY PLAIN
file path: N/A
file offset: 55
num of values: 2
total compressed size (in bytes): 51
total uncompressed size (in bytes): 47
data page offset: 27
index page offset: N/A
dictionary page offset: 4
```

When I run my code it prints following:
```
Row group: 0
Read column 0: 0 values (0 levels): [0, 0, 0, 0, 0, 0, 0, 0]
! Skip column 1, unknown type: BOOLEAN
Read column 2: 0 values (0 levels): [0, 0, 0, 0, 0, 0, 0, 0]
Read column 3: 0 values (0 levels): [0, 0, 0, 0, 0, 0, 0, 0]
Read column 4: 0 values (0 levels): [0, 0, 0, 0, 0, 0, 0, 0]
! Skip column 5, unknown type: INT64
! Skip column 6, unknown type: FLOAT
! Skip column 7, unknown type: DOUBLE
! Skip column 8, unknown type: BYTE_ARRAY
! Skip column 9, unknown type: BYTE_ARRAY
! Skip column 10, unknown type: INT96
```

All returned buffers are empty.

Turns out that there are several issues with reading values: when reading dictionary page we should not increment number of seen values, because they are just for dictionary, we have not read actual values yet, and also encoding assignment for data page .

diff contains updated `test_file_reader` test:
```diff
diff --git a/src/column/reader.rs b/src/column/reader.rs
index 9d36918..cea0bc5 100644
--- a/src/column/reader.rs
+++ b/src/column/reader.rs
@@ -213,9 +213,11 @@ impl<'a, T: DataType> ColumnReaderImpl<'a, T> where T: 'static {
                 self.def_level_decoder = Some(def_decoder);
               }
 
-              if encoding == Encoding::PLAIN_DICTIONARY {
-                encoding == Encoding::RLE_DICTIONARY;
-              }
+              let encoding = if encoding == Encoding::PLAIN_DICTIONARY {
+                Encoding::RLE_DICTIONARY
+              } else {
+                encoding
+              };
 
               let decoder =
                 if encoding == Encoding::RLE_DICTIONARY {
diff --git a/src/file/reader.rs b/src/file/reader.rs
index ea7871d..5863f57 100644
--- a/src/file/reader.rs
+++ b/src/file/reader.rs
@@ -308,7 +308,6 @@ impl PageReader for SerializedPageReader {
           assert!(page_header.dictionary_page_header.is_some());
           let dict_header = page_header.dictionary_page_header.as_ref().unwrap();
           let is_sorted = dict_header.is_sorted.unwrap_or(false);
-          self.seen_num_values += dict_header.num_values as i64;
           Page::DictionaryPage {
             buf: ByteBufferPtr::new(buffer), num_values: dict_header.num_values as u32,
             encoding: Encoding::from(dict_header.encoding), is_sorted: is_sorted
@@ -396,7 +395,7 @@ mod tests {
     let mut page_reader_0: Box<PageReader> = page_reader_0_result.unwrap();
     let mut page_count = 0;
     while let Ok(Some(page)) = page_reader_0.get_next_page() {
-      let is_dict_type = match page {
+      let is_expected_page = match page {
         Page::DictionaryPage{ buf, num_values, encoding, is_sorted } => {
           assert_eq!(buf.len(), 32);
           assert_eq!(num_values, 8);
@@ -404,12 +403,22 @@ mod tests {
           assert_eq!(is_sorted, false);
           true
         },
-        _ => false
+        Page::DataPage { buf, num_values, encoding, def_level_encoding, rep_level_encoding } => {
+          assert_eq!(buf.len(), 11);
+          assert_eq!(num_values, 8);
+          assert_eq!(encoding, Encoding::PLAIN_DICTIONARY);
+          assert_eq!(def_level_encoding, Encoding::RLE);
+          assert_eq!(rep_level_encoding, Encoding::BIT_PACKED);
+          true
+        },
+        _ => {
+          false
+        }
       };
-      assert!(is_dict_type);
+      assert!(is_expected_page);
       page_count += 1;
     }
-    assert_eq!(page_count, 1);
+    assert_eq!(page_count, 2);
   }
 
   fn get_test_file<'a>(file_name: &str) -> fs::File {
```

With these changes I get following output - looks correct:
```
Row group: 0
Read column 0: 2 values (0 levels): [6, 7, 0, 0, 0, 0, 0, 0]
! Skip column 1, unknown type: BOOLEAN
Read column 2: 2 values (0 levels): [0, 1, 0, 0, 0, 0, 0, 0]
Read column 3: 2 values (0 levels): [0, 1, 0, 0, 0, 0, 0, 0]
Read column 4: 2 values (0 levels): [0, 1, 0, 0, 0, 0, 0, 0]
! Skip column 5, unknown type: INT64
! Skip column 6, unknown type: FLOAT
! Skip column 7, unknown type: DOUBLE
! Skip column 8, unknown type: BYTE_ARRAY
! Skip column 9, unknown type: BYTE_ARRAY
! Skip column 10, unknown type: INT96
```

### PR DESCRIPTION
**Pull request by sadikovi**
This PR fixes reading columns with PLAIN_DICTIONARY encoding. 

Before the patch, reading of such columns would fail with unsupported encoding and/or would return empty buffer.
Example of column that returns empty buffer:
```
column type: INT32
column path: "id"
encodings: RLE PLAIN_DICTIONARY PLAIN
file path: N/A
file offset: 55
num of values: 2
total compressed size (in bytes): 51
total uncompressed size (in bytes): 47
data page offset: 27
index page offset: N/A
dictionary page offset: 4
```

Patch removes `seen_num_values` increment for dictionary pages, because they are usually followed by data page that contains "values"/indices. Also fixed encoding reassignment.

Updated unittest to reflect these changes + tested manually.

Fixes #10.

### CODE REVIEW

**Comment by sadikovi** [comment]
@sunchao please review. Thanks!

File: src/column/reader.rs [hunk 1]
```diff
@@ -213,9 +213,11 @@ impl<'a, T: DataType> ColumnReaderImpl<'a, T> where T: 'static {
                 self.def_level_decoder = Some(def_decoder);
               }
 
-              if encoding == Encoding::PLAIN_DICTIONARY {
-                encoding == Encoding::RLE_DICTIONARY;
-              }
+              let encoding = if encoding == Encoding::PLAIN_DICTIONARY {
```
Comment (sunchao) [inline src/column/reader.rs]: Can this be:
```
if encoding == Encoding::PLAIN_DICTIONARY {
  encoding = Encoding::RLE_DICTIONARY
}
```
? It is less verbose this way.

**Comment by sadikovi** [comment]
@sunchao updated code, so now I just reassign encoding like you suggested, though I have to use `mut encoding` in match statement for page. Is this okay?

**Comment by sunchao** [comment]
@sadikovi Yes I think it should be OK: the page is consumed upon usage so this modification won't be an issue.

**Pull request merged**